tokio-stream/src/stream_ext.rs

```diff
@@ -1005,6 +1007,62 @@ pub trait StreamExt: Stream {
     {
         throttle(duration, self)
     }
+
+    /// Batches the items in the given stream using a maximum duration and size for each batch.
+    ///
+    /// This stream returns the next batch of items in the following situations:
+    ///  1. The inner stream has returned at least `max_size` many items since the last batch.
+    ///  2. The time since the first element of a batch is greater than the given duration.
+    ///  3. The end of the stream is reached.
+    ///
+    /// The length of the returned vector is never empty or greater than the maximum size. Empty batches
+    /// will not be emitted if no items are received upstream.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if `max_size` is zero
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use std::time::Duration;
+    /// use tokio::time;
+    /// use tokio_stream::{self as stream, StreamExt};
+    /// use futures::FutureExt;
+    ///
+    /// #[tokio::main]
+    /// # async fn _unused() {}
+    /// #[tokio::main(flavor = "current_thread", start_paused = true)]
+    /// async fn main() {
```

Comment:
You need another hashtag on the second `tokio::main` here to make it look better in the documentation. Please try building the documentation locally to see how it looks. You can do so with this command:
```
RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --all-features
```

Reply:
ah, thanks! built and lgtm now